### DISCUSSION TITLE
Include original filename when importing archives

### DISCUSSION THREAD

**Discussion by haakobja** [beetbox/beets#4846] [Ideas]
During my last import I discovered that beets only display the tempdir directory name, not the original name of the imported zip-file. This makes it difficult to determine which albums must be re-imported due to failure to look up missing metadata.

Below is the stdout from Beets:

```
> beet --config=$HOME/mount/Music/.beets/config --directory=$HOME/mount/Music --library=$HOME/mount/Music/.beets/library.db import --set source=bandcamp -q -l $HOME/Music/bandcamp-import.log -i Bandcamp/*

/tmp/tmp81wjfvni (10 items)
Tagging:
    Austin Wintory & London Symphony Orchestra - Traveler - A Journey Symphony
URL:
    https://musicbrainz.org/release/f6bdb4fe-8e8d-4da8-b12c-fabbdbb48d8e
(Similarity: 100.0%) (Digital Media, 2022, XW, T-65b Records)
 * I Was Born For This -> I Was Born for This
This album is already in the library!
Skipping.

/tmp/tmpqermxk7q (20 items)
Tagging:
    Austin Wintory - Absolver
URL:
    https://austinwintory.bandcamp.com/album/absolver
(Similarity: 100.0%) (bandcamp, Digital Media, 2017, US, Austin Wintory)
This album is already in the library!
Skipping.

/tmp/tmpf_bpegr0 (18 items)
Tagging:
    Austin Wintory - ABZÛ
URL:
    https://musicbrainz.org/release/b5f11e80-8034-4eb6-a4fa-2d970b6bc4f1
(Similarity: 100.0%) (2x12" Vinyl, 2017, US, iam8bit, 8BIT-8047, Oceanic blue vinyl)
12" Vinyl 1
12" Vinyl 2
 * Chaos, the mother -> Chaos, the Mother
This album is already in the library!
Skipping.

/tmp/tmp_6vf5bfg (12 items)
Tagging:
    Austin Wintory - AGOS: A Game of Space
URL:
    https://austinwintory.bandcamp.com/album/agos-a-game-of-space-2
(Similarity: 100.0%) (bandcamp, Digital Media, 2020, US, Austin Wintory)
This album is already in the library!
Skipping.
```

And here's the log I got from beets:
```
> tail -f Music/bandcamp-import.log 
import started Fri Jul 14 11:07:25 2023
duplicate-skip /tmp/tmp81wjfvni
duplicate-skip /tmp/tmpqermxk7q
duplicate-skip /tmp/tmpf_bpegr0
duplicate-skip /tmp/tmp_6vf5bfg
duplicate-skip /tmp/tmp1d86c_2w
duplicate-skip /tmp/tmpy3t963qk
duplicate-skip /tmp/tmp66y2jkuy
duplicate-skip /tmp/tmpnidvpyd6
skip /home/user/mount/Original musikk/Bandcamp/Austin Wintory - Counter Strike- Global Offensive - Mocha Petal.flac
```

It would be nice to enable Beets to display the name of the imported zip-file in both the log and the stdout. Given that I ran this import as a quiet import, I may not check the stdout when I instruct beets to output an import log.

I recon this would be a nice-to-have for other users as well

*EDIT:* Removed username from beets log

**Comment by haakobja** [comment]
I guess the easiest way to do this is to include the zip-filename in the temporary directory name

**Comment by haakobja** [comment]
**New example**

*stdout*
```
/tmp/tmpjqrloi2y (19 items)
Skipping.

/tmp/tmpe_dh80jy (28 items)
Tagging:
    Various Artists - Sunset
URL:
    https://austinwintory.bandcamp.com/album/sunset
(Similarity: 100.0%) (bandcamp, Digital Media, 2015, US, Austin Wintory)

/tmp/tmpkrh5zrdz (6 items)
Tagging:
    Zweihänder - Daemon Slayer
URL:
    https://musicbrainz.org/release/98d868fd-5fa9-4a89-b946-662e74d7d3e8
(Similarity: 100.0%) (Digital Media, 2017, XW)
This album is already in the library!
Skipping.

``` 

*Log*:

```
skip /tmp/tmpjqrloi2y
duplicate-skip /tmp/tmpkrh5zrdz
``` 

As shown in the log, the first zip import was skipped, the second was imported and the third was detected as duplicate. Is there a way to get the name of the imported zip-file? Nether stdout or the log gives any clue to the name of the imported zip-file

**Comment by sampsyo** [comment]
This all makes sense as a worthy addition! TBH I don't immediately know how we'd accomplish it, but it would be best to just show the original zip file's path there, rather than forcing the temporary directory to have a reminiscent name. Maybe this should be converted into a feature request ticket?

**Reply by haakobja** [reply]
I agree that this should be converted into a feature request ticket as I think it would make it easier to identify albums skipped during import.
